dcc-portal-ui/app/scripts/compounds/js/compounds.js

```diff
@@ -361,8 +361,8 @@ angular.module('icgc.compounds.services', ['icgc.genes.models'])
             _compoundEntity = _compoundEntityFactory(compound.plain());
             defer.resolve(_self);
           }, function(){
-            Notify.hide();
-            $state.go('404', {page: 'compound', id: compoundId});
+            Page.stopWork();
+            $state.go('404', {page: 'compound', id: compoundId, url: '/compound/:id'}, {location: false});
```

Comment:
Also, we should be checking for 404 some how. What if it was another error code?